### PR TITLE
[PATH] fix #4810

### PR DESCRIPTION
**Pull request by sliptonic**
use subprocess.Popen() to avoid executing arbitrary code

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [ ]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by luzpaz** [comment]
@sliptonic is it mentioned anywhere in the docs that `asciidoctor` is a recommended dependency for Path WB?

File: src/Mod/Path/PathScripts/PathSanity.py [hunk 1]
```diff
@@ -464,7 +465,9 @@ def __report(self, data):
             )
 
         try:
-            result = os.system("asciidoctor {} -o {}".format(reportraw, reporthtml))
+            result = subprocess.Popen(
+                "asciidoctor {} -o {}".format(reportraw, reporthtml)
```
Comment (chennes) [inline src/Mod/Path/PathScripts/PathSanity.py:469]: Popen should take an array of arguments, not a pre-formatted string. The example from the manual is:
```
Popen(["/usr/bin/git", "commit", "-m", "Fixes a bug."])
```

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5306">pipeline status</a> for feature branch [PR_5306](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5306). Pipeline [439265238](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/439265238) was triggered at [debec19](https://github.com/FreeCAD/FreeCAD/pull/5306/commits/debec1913dda1dead5e5bbb70b31829cbd19e6d1). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/Path/PathScripts/PathSanity.py (hunk 1 above)
Comment (sliptonic) [reply src/Mod/Path/PathScripts/PathSanity.py]: Well that explains a lot.  resolved.

**Comment by sliptonic** [comment]
> @sliptonic is it mentioned anywhere in the docs that `asciidoctor` is a recommended dependency for Path WB?

https://wiki.freecadweb.org/Path_Sanity#Note_an_ASCIIDOC_and_Asciidoctor

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5306">pipeline status</a> for feature branch [PR_5306](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5306). Pipeline [439824097](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/439824097) was triggered at [a65dbc6](https://github.com/FreeCAD/FreeCAD/pull/5306/commits/a65dbc6f8296562a12407a36f4931a80bbb628b7). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Pull request merged**